components/automate-ui/src/app/entities/layout/layout-sidebar.service.ts

```diff
@@ -1,294 +1,304 @@
-import { Injectable } from '@angular/core';
+import { Injectable, OnInit, OnDestroy } from '@angular/core';
 import { Store } from '@ngrx/store';
-import { Observable } from 'rxjs';
+import { Subject, Observable, BehaviorSubject } from 'rxjs';
+import { forEach } from 'lodash';
 
 import { NgrxStateAtom } from 'app/ngrx.reducers';
 import { isProductDeployed } from 'app/staticConfig';
 import { isIAMv2 } from 'app/entities/policies/policy.selectors';
 import { FeatureFlagsService } from 'app/services/feature-flags/feature-flags.service';
 import { clientRunsWorkflowEnabled } from 'app/entities/client-runs/client-runs.selectors';
 import * as fromClientRuns from 'app/entities/client-runs/client-runs.reducer';
-
-import { MenuItemGroup } from './layout.model';
+import { UpdateSidebars } from './layout.actions';
+import { Sidebars } from './layout.model';
 
 @Injectable({
     providedIn: 'root'
 })
-export class LayoutSidebarService {
+export class LayoutSidebarService implements OnInit, OnDestroy {
     public applicationsFeatureFlagOn: boolean;
     public chefInfraServerViewsFeatureFlagOn: boolean;
     public isIAMv2$: Observable<boolean>;
-    public ServiceNowfeatureFlagOn: boolean;
-    private workflowEnabled: boolean;
-
+    public ServiceNowFeatureFlagOn: boolean;
+    private activeSidebar: string;
+    private workflowEnabled$: Observable<boolean>;
+    private isDestroyed = new Subject<boolean>();
+    private sidebars: Sidebars;
 
     constructor(
         private store: Store<NgrxStateAtom>,
         private clientRunsStore: Store<fromClientRuns.ClientRunsEntityState>,
         private featureFlagsService: FeatureFlagsService
     ) {
-        this.isIAMv2$ = this.store.select(isIAMv2);
-        this.chefInfraServerViewsFeatureFlagOn = this.featureFlagsService.getFeatureStatus('chefInfraServerViews');
         this.applicationsFeatureFlagOn = this.featureFlagsService.getFeatureStatus('applications');
-        this.ServiceNowfeatureFlagOn = this.featureFlagsService.getFeatureStatus('servicenow_cmdb');
-        this.clientRunsStore.select(clientRunsWorkflowEnabled).subscribe(
-            (workflowEnabled) => this.workflowEnabled = workflowEnabled
-        );
+        this.ServiceNowFeatureFlagOn = this.featureFlagsService.getFeatureStatus('servicenow_cmdb');
+        this.isIAMv2$ = this.store.select(isIAMv2);
+        this.workflowEnabled$ = this.clientRunsStore.select(clientRunsWorkflowEnabled);
     }
 
-    public getDashboardsSidebar(): MenuItemGroup[] {
-        return [{
-            name: 'Dashboards',
+    populateSidebar() {
+      const sidebars = {
```

Comment:
It is important to strongly type this to catches any typos, invalid assignments, or missing properties. I realize you will have to add a few values here and there to satisfy it, but I think it is worth it. 
```suggestion
      const sidebars: Sidebars = {
```
For instance, as soon as I tried, it revealed that L85 (`visible$: this.applicationsFeatureFlagOn`) has the wrong type.

Reply:
@msorens Updated change request 😊